translator/defaultKeyCase.go

```
@@ -74,6 +74,27 @@ func DefaultStringArrayCase(key string, defaultVal, input interface{}) (returnKe
 	return
 }
 
+func DefaultIntegralArrayCase(key string, defaultVal, input interface{}) (string, interface{}) {
+	returnKey, returnVal := DefaultCase(key, defaultVal, input)
+	if arrayVal, ok := returnVal.([]interface{}); ok {
+		intArrayVal := []int{}
+		for _, v := range arrayVal {
+			if floatVal, ok := v.(float64); ok {
+				intVal := int(floatVal)
+				if IsValidEventIDs(intVal) {
+					intArrayVal = append(intArrayVal, intVal)
+				}
```

Since this function is generic for integer arrays, I would keep this validation outside of this function. Separate the process of converting the configuration value to an integer array and the processing of validating the integer array. It might be slightly less efficient (double iterating), but it keeps event ID validation logic separate from JSON parsing logic. It'll also let you report errors for individual values that are not valid event IDs instead of silently swallowing those issues.